packages/edit-site/src/components/page-templates/fields.js

```diff
@@ -181,17 +172,20 @@ export const useThemeField = () => {
 	const activeTheme = useSelect( ( select ) =>
 		select( coreStore ).getCurrentTheme()
 	);
-	return {
-		label: __( 'Compatible Theme' ),
-		id: 'theme',
-		getValue: ( { item } ) => item.theme,
-		render: function Render( { item } ) {
-			if ( item.theme === activeTheme.stylesheet ) {
-				return <Badge intent="success">{ item.theme }</Badge>;
-			}
-			return <Badge intent="error">{ item.theme }</Badge>;
-		},
-	};
+	return useMemo(
+		() => ( {
+			label: __( 'Compatible Theme' ),
+			id: 'theme',
+			getValue: ( { item } ) => item.theme,
+			render: function Render( { item } ) {
+				if ( item.theme === activeTheme.stylesheet ) {
+					return <Badge intent="success">{ item.theme }</Badge>;
+				}
+				return <Badge intent="error">{ item.theme }</Badge>;
```

Comment:
Nit: we're creating 2 separate `<Badge />` components, when we could have made just the `intent` prop have a conditional value.